File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/composite/transport/TransportGroup.java
// TransportGroup.java

package com.bigboxcode.designpattern.composite.transport;

import java.util.ArrayList;
import java.util.List;

public class TransportGroup implements Transport {
    private List<Transport> transportList = new ArrayList<Transport>();

    @Override
    public void start() {
        for (Transport transport: transportList) {
            transport.start();
        }
    }

    @Override
    public void operate() {
        for (Transport transport: transportList) {
            transport.operate();
        }
    }

    @Override
    public void stop() {
        for (Transport transport: transportList) {
            transport.stop();
        }
    }

    public void addTransport(Transport transport) {
        transportList.add(transport);
    }

    public void removeTransport(Transport transport) {
        transportList.remove(transport);
    }
}

File: design-patterns/php/composite/transportlist/TransportGroup.php
<?php
// TransportGroup.php

namespace BigBoxCode\DesignPattern\Composite\TransportList;

class TransportGroup implements Transport {
    private array $transportList = [];

    public function start(): void {
        foreach ($this->transportList as $transport) {
            $transport->start();
        }
    }

    public function operate(): void {
        foreach ($this->transportList as $transport) {
            $transport->operate();
        }
    }

    public function stop(): void {
        foreach ($this->transportList as $transport) {
            $transport->stop();
        }
    }

    public function addTransport(Transport $transport): void {
        array_push($this->transportList, $transport);
    }

    public function removeTransport(Transport $transport): void {
        $elemIndex = array_search($transport, $this->transportList);

        if ($elemIndex !== false) {
            unset($this->transportList[$elemIndex]);
        }
    }
}